uaclient/apt.py

```diff
@@ -490,10 +490,20 @@ def is_installed(pkg: str) -> bool:
     return pkg in get_installed_packages()
 
 
-def get_installed_packages() -> List[str]:
+def get_installed_packages(include_versions: bool = False) -> List[str]:
     out, _ = system.subp(["apt", "list", "--installed"])
     package_list = out.splitlines()[1:]
-    return [entry.split("/")[0] for entry in package_list]
+    pkg_names = [entry.split("/")[0] for entry in package_list]
+    if include_versions:
+        pkg_versions = [entry.split(" ")[1:] for entry in package_list]
+        pkgs = []
+        for pkg in zip(pkg_names, pkg_versions):
+            arch = pkg[1][1]
+            arch = "" if arch == "all" else ":" + arch
+            version = pkg[1][0]
+            pkgs.append("{}{}\t{}".format(pkg[0], arch, version))
+        return pkgs
+    return pkg_names
```

Comment:
Sorry for disagreeing here, but I think this should be two different functions. Generally I don't think the value of a parameter should meaningfully change the type of the return value.

I suggest:
```python
PackageVersion = NamedTuple("PackageVersion", [("name", str), ("version", str)])
def get_installed_packages_with_versions() -> List[PackageVersion]:
    out, _ = system.subp(["apt", "list", "--installed"])
    package_list = out.splitlines()[1:]
    return [
        PackageVersion(
            name=entry.split("/")[0],
            version=entry.split(" ")[1]
        ) for entry in package_list
    ]

def get_installed_packages() -> List[str]:
    return [package.name for package in get_installed_packages_with_versions()]
```

Reply:
I was thinking about this part, thanks for this different approach
I thought my approach was abit brittle.